common/types/data_connections.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiRadioGroupOption } from '@elastic/eui';
+
+export interface PermissionsFlexItem {
+  roles: Array<{ label: string }>;
+  selectedRoles: Array<{ label: string }>;
+  selectedRadio: string;
+  setSelectedRoles: (selectedRoles: Array<{ label: string }>) => void;
+  setSelectedRadio: (selectedRadio: string) => void;
```

Comment:
Minor, but Ideally types here should be something like: `React.Dispatch<React.SetStateAction<Array<{ label: string }>>`

Reply:
Done in: [fc96304](https://github.com/opensearch-project/dashboards-observability/pull/992/commits/fc96304362b085696c2e5df8e7c638b69c142180), thanks! I didn't know this